src/front/glsl/mod.rs

```diff
@@ -17,11 +21,19 @@ mod token;
 mod types;
 mod variables;
 
-pub fn parse_str(source: &str, entry: &str, stage: ShaderStage) -> Result<Module, ParseError> {
-    log::debug!("------ GLSL-pomelo ------");
-
+pub fn parse_str(
+    source: &str,
+    entry: &str,
+    stage: ShaderStage,
+    defines: Vec<(String, String)>,
```

Comment:
Why not get it in the format we expect, right away (i.e. `FastHashMap<String, String>`)?

Reply:
How about `Option<FastHashMap<String, String>>`? Somewhat easier on your fingers in the cases where you don't really car about initial defines (and can just write `None`)

Reply:
Hmmm after thinking some more about this, wouldn't this be imposing our `FastHashMap` on the user, as this is exposed in the external interface? Or is `FastHashMap` with `fxhash` so widely used anyway that this won't be a problem?